src/combat.cpp

```diff
@@ -701,8 +702,7 @@ void Combat::doCombat(Creature* caster, Creature* target) const
 		}
 	} else {
 		if (!params.aggressive || (caster != target && Combat::canDoCombat(caster, target) == RETURNVALUE_NOERROR)) {
-			SpectatorVec spectators;
-			g_game.map.getSpectators(spectators, target->getPosition(), true, true);
+			Spectators spectators = g_game.map.getSpectators(target->getPosition(), true, true);
```

Comment:
const Spectators& spectators ...

this applies to every single occurrence of similar code.

Reply:
Thanks!